File: py/등굣길.py
def solution(m, n, puddles):
    visit = [[0]*(m+1) for _ in range(n+1)]
    visit[1][1] = 1
    for y in range(1,n+1):
        for x in range(1,m+1):
            if [x,y] in puddles or [1,1]==[y,x]:
                continue
            visit[y][x]=(visit[y-1][x]+visit[y][x-1])%1000000007
    return visit[n][m]

File: 프로그래머스/3/42898. 등굣길/등굣길.java
class Solution {
    public int solution(int m, int n, int[][] puddles) {
        
        int[][] map = new int[n][m];
        
        
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                map[i][j]=0;
            }
        }
        for(int[] water : puddles) map[water[1]-1][water[0]-1]=123456789;
        map[0][0]=1;
        
        int[] dy={-1,0};
        int[] dx={0,-1};
        
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                
                if(map[i][j]!=0) continue;
                
                for(int e=0;e<2;e++){
                    int y = i+dy[e];
                    int x = j+dx[e];
                    if(y<0||x<0||map[y][x]==123456789) continue;
                    map[i][j]=(map[y][x]+map[i][j])%1000000007;
                }
            }
        }
        
        
        return map[n-1][m-1];
        
        
        
    }
}